src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -441,6 +456,206 @@ try
+    }
+}
+
+// Routine Description:
+// - Updates our internal map of font axes with the given axes
+// - NOTE TO CALLER: Make sure to call _BuildFontRenderData after calling this for the axes changes
+//   to take place
+// Arguments:
+// - axes - the axes to update our map with
+void DxFontRenderData::_SetAxes(const std::unordered_map<std::wstring_view, int32_t>& axes)
+{
+    _axesVector.clear();
+
+    // Update our axis map with the provided axes
+#pragma warning(suppress : 26445) // the analyzer doesn't like reference to string_view
+    for (const auto& [axis, value] : axes)
+    {
+        if (axis.length() == TAG_LENGTH)
+        {
+            const auto dwriteTag = DWRITE_MAKE_FONT_AXIS_TAG(til::at(axis, 0), til::at(axis, 1), til::at(axis, 2), til::at(axis, 3));
+            _axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ dwriteTag, gsl::narrow_cast<float>(value) });
+        }
+    }
+}
+
+// Method Description:
+// - Converts a DWRITE_FONT_STRETCH enum into the corresponding float value to
+//   create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontStretch: the old DWRITE_FONT_STRETCH enum to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontStretch
+float DxFontRenderData::FontStretchToWidthAxisValue(DWRITE_FONT_STRETCH fontStretch) noexcept
+{
+    if (gsl::narrow_cast<size_t>(fontStretch) > fontStretchEnumToVal.size())
+    {
+        fontStretch = DWRITE_FONT_STRETCH_NORMAL;
+    }
+
+    return til::at(fontStretchEnumToVal, fontStretch);
+}
+
+// Method Description:
+// - Converts a DWRITE_FONT_STYLE enum into the corresponding float value to
+//   create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontStyle: the old DWRITE_FONT_STYLE enum to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontStyle
+float DxFontRenderData::FontStyleToSlantFixedAxisValue(DWRITE_FONT_STYLE fontStyle) noexcept
+{
+    // Both DWRITE_FONT_STYLE_OBLIQUE and DWRITE_FONT_STYLE_ITALIC default to having slant.
+    // Though an italic font technically need not have slant (there exist upright ones), the
+    // vast majority of italic fonts are also slanted. Ideally the slant comes from the
+    // 'slnt' value in the STAT or fvar table, or the post table italic angle.
+
+    if (gsl::narrow_cast<size_t>(fontStyle) > fontStyleEnumToVal.size())
+    {
+        fontStyle = DWRITE_FONT_STYLE_NORMAL;
+    }
+
+    return til::at(fontStyleEnumToVal, fontStyle);
+}
+
+// Method Description:
+// - Converts a float fontSize into the corresponding float value to create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontSize: the old float value to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontSize
+float DxFontRenderData::DIPsToPoints(const float fontSize) noexcept
+{
+    return fontSize * (72.0f / 96.0f);
+}

```

Comment:
nit: technically, this is just... something you could make a local constexpr function. It doesn't need to be part of the DxFontRenderData class because it doesn't rely on any of its members or any of its other implementation details.

Reply:
Makes sense, thanks